### ISSUE TITLE
dcos-cli produces Mach-O executable

### ISSUE THREAD

**Issue by jkt628** [Homebrew/linuxbrew-core#12761]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?

**this is Linux, getting rid of all the config scripts is not really advisable.**

- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

https://gist.github.com/6530c59f8f7dfb0f12805b19dd8e1542

- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

`brew install dcos-cli`

- What happened (include command output)

```
==> Downloading https://github.com/dcos/dcos-cli/archive/0.8.0.tar.gz
==> Downloading from https://codeload.github.com/dcos/dcos-cli/tar.gz/0.8.0
    # -#O=-#   #
==> make darwin
🍺  /home/linuxbrew/.linuxbrew/Cellar/dcos-cli/0.8.0: 3 files, 11.3MB, built in 5 seconds
```
`file /home/linuxbrew/.linuxbrew/Cellar/dcos-cli/0.8.0/bin/dcos`
```
/home/linuxbrew/.linuxbrew/Cellar/dcos-cli/0.8.0/bin/dcos: Mach-O 64-bit x86_64 executable
```

- What you expected to happen

`make linux` or some such

- Step-by-step reproduction instructions (by running `brew install` commands)

```bash
brew install dcos-cli
file /home/linuxbrew/.linuxbrew/Cellar/dcos-cli/0.8.0/bin/dcos
```

**Comment by jkt628** [comment]
any guesses?

**Comment by issyl0** [comment]
@jkt628 Thanks for reporting this! I was able to reproduce the issue, and have submitted a fix in the PR linked above.

**Comment by iMichka** [comment]
Fixed by #13237
